commons/src/main/java/com/powsybl/commons/extensions/Extendable.java

```diff
@@ -49,4 +49,39 @@
      * @return all extensions associated to this extendable object.
      */
     <E extends Extension<O>> Collection<E> getExtensions();
+
+    /**
+     * Returns a name that is used to find matching {@link ExtensionAdderProvider}s
+     * when selecting implementations of extensions in {@link #newExtension}. This
+     * is meant to be overriden by extendables when multiple implementations exist.
+     *
+     * @return the name
+     */
+    default String getImplementationName() {
+        return "Default";
+    }
+
+    /**
+     * Returns an extensionAdder to build and add an extension for this extendable.
+     *
+     * The extension implementation is selected at runtime based on matching the
+     * {@link #getImplementationName} of this extendable to the
+     * {@link ExtensionAdderProvider#getImplementationName} of a provider.
+     * Implementations are loaded with java's {@link java.util.ServiceLoader} using
+     * the ExtensionAdderProvider interface.
+     *
+     * This is not meant to be overriden.
```

Comment:
Could you declare this method final?

Reply:
No, you can't have a final method in an interface

Reply:
But you agree that we do not want someone to overwrite it?